File: elixir/raindrops/raindrops.exs
defmodule Raindrops do
  @doc """
  Returns a string based on raindrop factors.

  - If the number contains 3 as a prime factor, output 'Pling'.
  - If the number contains 5 as a prime factor, output 'Plang'.
  - If the number contains 7 as a prime factor, output 'Plong'.
  - If the number does not contain 3, 5, or 7 as a prime factor,
    just pass the number's digits straight through.
  """
  @spec convert(pos_integer) :: String.t
  def convert(number) do

    plingplangplong = [ 
      "#{if 0 == rem(number, 3), do: "Pling"}", 
      "#{if 0 == rem(number, 5), do: "Plang"}", 
      "#{if 0 == rem(number, 7), do: "Plong"}" 
    ] |> Enum.join
      
    case plingplangplong do
      "" -> number |> to_string
      _ -> plingplangplong
    end
  end
end

File: teams/microsoft-cse-golang-study-group-summer-2019/go/raindrops/raindrops.go
package raindrops

import (
	"strconv"
	"strings"
)

func Convert(n int) string {
	sli := make([]string, 3)
	if n%3 == 0 {
		sli = append(sli, "Pling")
	}
	if n%5 == 0 {
		sli = append(sli, "Plang")
	}
	if n%7 == 0 {
		sli = append(sli, "Plong")
	}
	plingPlangPlong := strings.Join(sli[:], "")

	if plingPlangPlong != "" {
		return plingPlangPlong
	}
	return strconv.Itoa(n)
}
